server/channels/store/sqlstore/post_store.go

```diff
@@ -1110,17 +1130,34 @@ func (s *SqlPostStore) permanentDeleteAllCommentByUser(userId string) (err error
 // Permanently deletes all comments by user,
 // cleans up threads (removes said user from participants and decreases reply count),
 // permanent delete all root posts by user,
+// permanent delete all reactions by user,
 // and delete threads and thread memberships for those root posts
+// Note that this method can potentially take a very long time, and is not meant
+// to be called synchronously, but only via jobs.
 func (s *SqlPostStore) PermanentDeleteByUser(rctx request.CTX, userId string) error {
 	// First attempt to delete all the comments for a user
-	if err := s.permanentDeleteAllCommentByUser(userId); err != nil {
-		return err
+	for {
+		results := []postIds{}
+		err := s.GetReplica().Select(&results, "Select Id, RootId FROM Posts WHERE UserId = ? AND RootId != '' LIMIT 10", userId)
```

Comment:
I guess it's an unlikely scenario with the default interval (1 second), but what would happen if the replica returned already deleted records from a previous iteration?

nit: `Select` looks super weird :p

Reply:
Nothing much would happen. Only the ReplyCount number might be off for that time, but when it eventually syncs, it would be correct.